File: 프로그래머스/lv2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
class Solution {
    public int[] solution(int n, long left, long right) {
        int len = (int) (right - left + 1);
        int[] answer = new int[len];

        for (int i = 0; i < len; i++) {
            long num = left + i;
            int row = (int) (num / n);
            int col = (int) (num % n);

            if (col >= row) {
                answer[i] = col + 1;
            } else {
                answer[i] = row + 1;
            }
        }

        return answer;
    }
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.py
def solution(n, left, right):
    length = right - left + 1
    answer = [0] * length

    for i in range(length):
        num = left + i
        row = num // n
        col = num % n
        answer[i] = max(row + 1, col + 1)

    return answer